src/cmd/flash.js

```diff
@@ -135,6 +144,18 @@ class FlashCommand {
 			throw new VError(ensureError(err), 'Error writing firmware');
 		});
 	}
+
+	_dropModuleInfo(binary) {
+		// Creates a temporary binary with module info stripped out and returns the path to it
+
+		return when.promise((resolve, reject) => {
+			const rStream = fs.createReadStream(binary, { start: ModuleInfo.HEADER_SIZE });
```

Comment:
Can you speak of the decision to stripping the module header in the CLI instead of doing it in Device OS? This is going to make it difficult to flash this file with raw dfu-util without using the CLI. I'm not opposed to the idea of doing this in the CLI, I'm just wondering what the tradeoff was.

Reply:
The DeviceOS handles that when the binary is flashed over serial or OTA, it's impossible to do over DFU, because all `dfu-util` does is writes a binary to a particular location and that's it and SoftDevice needs to be at a specific location without any kind of prefix (module header in our case).

Reply:
We could alternatively do something like putting the header at the end of the file before the footer, updating heuristics to find it (https://github.com/particle-iot/binary-version-reader/blob/master/lib/HalModuleParser.js#L202), and also updating the DeviceOS to support that, but it honestly felt like a hack.

Reply:
Thanks for the explanation :shipit: